apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/springwebmvc/SpringServiceNameInstrumentation.java

```diff
@@ -67,50 +67,47 @@ public String getAdviceClassName() {
 
     public static class SpringServiceNameAdvice {
 
-        private static final Logger logger = LoggerFactory.getLogger(SpringServiceNameAdvice.class);
-
         @Advice.OnMethodExit(suppress = Throwable.class, inline = false)
         public static void afterInitPropertySources(@Advice.This WebApplicationContext applicationContext) {
+            // avoid having two service names for a standalone jar
+            // one based on Implementation-Title and one based on spring.application.name
+            if (!ServiceInfo.autoDetected().isMultiServiceContainer()) {
```

Comment:
Some background:
The risk of that is that if we don't properly detect that we're running on an application server/servlet container. In that case, we wouldn't be able to determine the service name based on `spring.application.name`. That's why I didn't add the check in `Tracer.overrideServiceInfoForClassLoader` but made it specific to this instrumentation. This way, we at least can still override based on the servlet context path and the display-name. It's quite unlikely that a standalone application that's using embedded Tomcat, for example, sets a display-name or even a servlet context path. Therefore, if any of these are set, that's a pretty good indication that we're running in a container and not in a standalone jar.